scanner/src/funTest/kotlin/experimental/DefaultNestedProvenanceResolverFunTest.kt

```diff
@@ -0,0 +1,202 @@
+                        revision = "fcea94bab5835172e826afddb9f6427274c983b9"
+                    ),
+                    resolvedRevision = "fcea94bab5835172e826afddb9f6427274c983b9"
+                )
+
+                val tree = resolver.resolveNestedProvenance(provenance)
+
+                tree.root shouldBe provenance
+                tree.subRepositories should containExactly(
+                    "commons-text" to RepositoryProvenance(
+                        vcsInfo = VcsInfo(
+                            type = VcsType.GIT,
+                            url = "https://github.com/apache/commons-text.git",
+                            revision = "7643b12421100d29fd2b78053e77bcb04a251b2e"
+                        ),
+                        resolvedRevision = "7643b12421100d29fd2b78053e77bcb04a251b2e"
+                    ),
+                    "test-data-npm" to RepositoryProvenance(
+                        vcsInfo = VcsInfo(
+                            type = VcsType.GIT,
+                            url = "https://github.com/oss-review-toolkit/ort-test-data-npm.git",
+                            revision = "ad0367b7b9920144a47b8d30cc0c84cea102b821"
+                        ),
+                        resolvedRevision = "ad0367b7b9920144a47b8d30cc0c84cea102b821"
+                    ),
+                    "test-data-npm/isarray" to RepositoryProvenance(
+                        vcsInfo = VcsInfo(
+                            type = VcsType.GIT,
+                            url = "https://github.com/juliangruber/isarray.git",
+                            revision = "63ea4ca0a0d6b0574d6a470ebd26880c3026db4a"
+                        ),
+                        resolvedRevision = "63ea4ca0a0d6b0574d6a470ebd26880c3026db4a"
+                    ),
+                    "test-data-npm/long.js" to RepositoryProvenance(
+                        vcsInfo = VcsInfo(
+                            type = VcsType.GIT, url = "https://github.com/dcodeIO/long.js.git",
+                            revision = "941c5c62471168b5d18153755c2a7b38d2560e58"
+                        ),
+                        resolvedRevision = "941c5c62471168b5d18153755c2a7b38d2560e58"
+                    )
+                )
+            }
+
+            "Find Git-Repo projects with recursive Git submodules".config(enabled = !Os.isWindows) {
+                val provenance = RepositoryProvenance(
+                    vcsInfo = VcsInfo(
+                        type = VcsType.GIT_REPO,
+                        url = "https://github.com/oss-review-toolkit/ort-test-data-git-repo.git",
+                        revision = "31588aa8f8555474e1c3c66a359ec99e4cd4b1fa",
+                        path = "manifest.xml"
+                    ),
+                    resolvedRevision = "31588aa8f8555474e1c3c66a359ec99e4cd4b1fa"
+                )
+
+                val tree = resolver.resolveNestedProvenance(provenance)
```

Comment:
This is still called "tree", not sure if it's intentional. Could be e.g. just "repo".